src/veins/modules/mobility/traci/subscriptionManagement/TraCISubscriptionManager.h

```diff
@@ -0,0 +1,218 @@
+     *
+     * @return list<TraCIPerson> a list of persons.
+     */
+    std::list<TraCIPerson> getUpdatedPersons();
+
+    /**
+     * Returns a set of persons that disappeared since the last time
+     * you called this method.
+     *
+     * @return set<string> a set of disappeared person ids.
+     */
+    std::set<std::string> getDisappearedPersons();
+
+    /**
+     * Returns a list of vehicle that are new or updated. It is up
+     * to the caller to decide which vehicle is updated and which is
+     * new.
+     *
+     * Multiple calls to this method will not yield the same result. It
+     * will always give you the new or updated vehicles since the last
+     * time you called this method.
+     *
+     * @return list<TraCIVehicle> a list of vehicles.
+     */
+    std::list<TraCIVehicle> getUpdatedVehicles();
+
+    /**
+     * Returns a set of vehicles that disappeared since the last time
+     * you called this method.
+     *
+     * @return set<string> a set of disappeared vehicle ids.
+     */
+    std::set<std::string> getDisappearedVehicles();
+
+
+    /**
+     * Get the ids of vehicles that started teleporting since
+     * the last time you called this method.
+     *
+     * Note: Multiple calls to this method will not yield the
+     * same result.
+     *
+     * @return list<string> a list of ids that started teleporting.
+     */
+    std::list<std::string> getStartedTeleporting();
+
+    /**
+     * Get the ids of vehicles that started parking since
+     * the last time you called this method.
+     *
+     * Note: Multiple calls to this method will not yield the
+     * same result.
+     *
+     * @return list<string> a list of ids that started parking.
+     */
+    std::list<std::string> getStartedParking();
+
+    /**
+     * Get the ids of vehicles that ended parking since
+     * the last time you called this method.
+     *
+     * Note: Multiple calls to this method will not yield the
+     * same result.
+     *
+     * @return list<string> a list of ids that ended parking.
+     */
+    std::list<std::string> getEndedParking();
+
+    /**
+     * Gives a list of traffic light subscription updates since the last
+     * time this method was called.
+     *
+     * @return std::list<TraCITrafficLight> list of traffic light updates.
+     */
+    std::list<TraCITrafficLight> getTrafficLightUpdates();
+
+    /**
+     * Subscribes to the following variables of a traffic light:
+     *      - current phase (0x28)
+     *      - current program (0x29)
+     *      - next switch (0x2d)
+     *      - red yellow green state (0x20)
+     *
+     * @param id of the traffic light to subscribe to.
+     */
+    void subscribeToTrafficLight(std::string id);
+
+    /**
+     * Initialize this manager with the given parameters to access the traci server.
+     */
+    void initialize(std::shared_ptr<TraCIConnection> connection, std::shared_ptr<TraCICommandInterface> commandInterface);
```

Comment:
As with the individual Managers, move this into the constructor an you never have to worry about uninirialized instances.

Reply:
see discussion above.

Reply:
Alright.